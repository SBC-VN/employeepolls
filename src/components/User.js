/**
* @description A single user row.  This is used in the LeaderBoard component.
* 
* @param - user - The user object to display.
*/

const User = ({user}) => {
    return (
        <div className = "user-row">
            <div className = "user-avatar-column">
                <img src={user.avatarURL} alt={`Avatar of ${user.name}`} className='avatar-small' />
                <span className = "span-inline"><h3>{user.id}</h3></span>                                
                <div className = "user-username">
                    {user.name}
                </div>
            </div>
            <div className ="user-activity-count" data-testid="answer-output">
                {Object.keys(user.answers).length}
            </div>
            <span className="span-columndivider"></span>
            <div className ="user-activity-count" data-testid="question-output">
                {user.questions.length}
            </div>  
        </div>
    )
};

export default User;