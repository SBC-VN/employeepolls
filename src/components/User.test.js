import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import User from './User';

let userInfo = {
    id: "sarahedo",
    name: "Sarah Edo",
    avatarURL: "../../images/snow.jpg",
    answers: {
        "8xf0y6ziyjabvozdd253nd": "optionOne",
        "6ni6ok3ym7mf1p33lnez": "optionOne",
        "am8ehyc8byjqgar0jgpub9": "optionTwo",
        "loxhs1bqm25b708cmbf3g": "optionTwo"
    },
    questions: ["8xf0y6ziyjabvozdd253nd", "6ni6ok3ym7mf1p33lnez"]
};

describe('User tests', () => {
    test('renders user row', () => {
        render(<User user={userInfo} />);
        screen.debug();
    });

    test('Snapshot test', () => {
        let component = render(<User user={userInfo} />);
        expect(component).toMatchSnapshot(); 
    });

    test('User info displays correctly', () => {
        let component = render(<User user={userInfo} />);
        expect(component.getByText("Sarah Edo")).toBeInTheDocument();
        expect(component.getByText("sarahedo")).toBeInTheDocument();
    });

    test('Answer/Quesion count displays correctly', () => {
        let component = render(<User user={userInfo} />);
        expect(component.getByTestId("answer-output")).toHaveTextContent("4");
        expect(component.getByTestId("question-output")).toHaveTextContent("2");
    });
});