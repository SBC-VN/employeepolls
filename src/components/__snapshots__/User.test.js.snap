// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User tests Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="user-row"
      >
        <div
          class="user-avatar-column"
        >
          <img
            alt="Avatar of Sarah Edo"
            class="avatar-small"
            src="../../images/snow.jpg"
          />
          <span
            class="span-inline"
          >
            <h3>
              sarahedo
            </h3>
          </span>
          <div
            class="user-username"
          >
            Sarah Edo
          </div>
        </div>
        <div
          class="user-activity-count"
          data-testid="answer-output"
        >
          4
        </div>
        <span
          class="span-columndivider"
        />
        <div
          class="user-activity-count"
          data-testid="question-output"
        >
          2
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="user-row"
    >
      <div
        class="user-avatar-column"
      >
        <img
          alt="Avatar of Sarah Edo"
          class="avatar-small"
          src="../../images/snow.jpg"
        />
        <span
          class="span-inline"
        >
          <h3>
            sarahedo
          </h3>
        </span>
        <div
          class="user-username"
        >
          Sarah Edo
        </div>
      </div>
      <div
        class="user-activity-count"
        data-testid="answer-output"
      >
        4
      </div>
      <span
        class="span-columndivider"
      />
      <div
        class="user-activity-count"
        data-testid="question-output"
      >
        2
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
